src/actions/start.ts

```diff
@@ -1,22 +1,42 @@
 import express from 'express';
-import { createLogger } from '@w3f/logger';
+import { createLogger, Logger } from '@w3f/logger';
 import { Config } from '@w3f/config';
-
 import { Subscriber } from '../subscriber';
 import { InputConfig } from '../types';
+import { WsMessageCenter } from "../messageCenter";
+
+export class StartAction {
+  private cfg: InputConfig;
+  private subscriber: Subscriber;
+  private logger: Logger;
+
+  public execute = async (cmd: { config: string }): Promise<void> => {
+
+    this._initInstanceVariables(cmd)
 
+    await this.subscriber.start();
 
-export async function startAction(cmd): Promise<void> {
-    const cfg = new Config<InputConfig>().parse(cmd.config);
+    const wsMC = new WsMessageCenter(this.cfg,this.logger,this.subscriber)
```

Comment:
I really love the idea of having the ability to set the handler! :100: But here the signatures look a bit confusing, `WsMessageCenter` requires `this.subscriber` in its constructor, and later the object created is set as a member of `this.subscriber`... Maybe the subscriber part that is required by WsMessageCenter could be refactored?

Reply:
I didn't want to have a circular dependency, so what I've done is:
- MessageCenter requires a subscriber in his constructor. MessageCenter can call the public methods of Subscriber.
- Subscriber has nothing to do with MessageCenter, but it has an handler function that needs to be defined from outside
- MessageCenter is providing just that function to the Subscriber.

I learnt this kind of approach working with React where it was really common acting like that, 
WDYT ??

Reply:
Sounds good :+1: